docs/source/guide/tasks.md

```diff
@@ -102,6 +103,119 @@ http://example.com/text.txt
 
 {% enddetails %}
 
+
+## How to retrieve data
+
+There are several steps to retrieve the data to display in the `Object` tag. The data retrieval is also used in [dynamic choices](/templates/serp_ranking.html) and [labels](/templates/inventory_tracking.html). Use the following parameters in the `Object` tag.
+
+### `value` (required)
+
+The `value` parameter represents the source of the data. It can be plain text or a step of complex data retrieval system. It can be denoted using the following forms:
+`value` (required)
+
+#### Variables 
+
+In most cases, the `Object` tag has the value with one variable (prefixed with a $) in it. For example, `<Audio value="$audio" ... />`. All variables  point to field in task data. For example, `$audio => task.data.audio`
+
+!!! attention "important"
+    In most cases, the `Object` tag has the value with one variable (prefixed with a $) in it. For example, `<Audio value="$audio" ... />`. All the variables point to a field in task data. For example, `$audio => task.data.audio`.
+
+
+#### Text
+
+The value parameter can be a string. It is useful for `Header` and `Text`. 
+
+Also, you can use the content of the tag as value. It is useful for descriptive text tags and is applied for `Label` and `Choice`.
+
+For example:
+
+```html
+<Header>Label audio:</Header>
+
+<Header value="Label only fully visible cars" />
+
+<Text name="instruction" value="Label only fully visible cars" />
+
+<Label>cat</Label>
+
+<Choice>other</Choice>
```

Comment:
remove empty lines here

Reply:
This implies by default there should be no line breaks within code segments.